lib/scanner/walk.go

```diff
@@ -247,44 +290,186 @@ func (w *walker) walkAndHashFiles(ctx context.Context, fchan, dchan chan protoco
 			return skip
 		}
 
-		if w.Matcher.Match(path).IsIgnored() {
-			l.Debugln("ignored (patterns):", path)
-			return skip
-		}
-
 		if !utf8.ValidString(path) {
+			select {
+			case fsChan <- fsWalkResult{
+				path: path,
+				info: nil,
+				err:  errors.New("Path isn't a valid utf8 string"),
+			}:
+			case <-ctx.Done():
+				return ctx.Err()
+			}
 			l.Warnf("File name %q is not in UTF8 encoding; skipping.", path)
 			return skip
 		}
 
 		path, shouldSkip := w.normalizePath(path)
 		if shouldSkip {
+			select {
```

Comment:
This little select is repeated often enough that it would be a nice `func walkError(ctx context.Context, dst chan WalkResult, path string, err error)` I think